s3backgrounddelete/s3backgrounddelete/object_recovery_msgbus_consumer.py

```diff
@@ -65,14 +67,27 @@ def _process_msg(self, msg):
             return False
         return True
 
+    def __internalconnect(self):
+        self._logger.debug("Instantiating S3MessageBus")
+        self._consumer = S3CortxMsgBus()
+        self._logger.debug("Connecting to S3MessageBus")
+        ret,msg = self._consumer.connect()
+        if ret:
+            ObjectRecoveryMsgbusConsumer.isMsgBusLoaded = True
+        else:
+            self._logger.error("connect failed {}".format(msg))
+            self._consumer = None
+        return ret
+
     def receive_data(self):
         """Initializes consumer, connects and receives messages from message bus"""
         try:
-            self._logger.debug("Instantiating S3MessageBus")
-            self._consumer = S3CortxMsgBus()
-            self._logger.debug("Connecting to S3MessageBus")
-            ret,msg = self._consumer.connect()
-            if ret:
+            if not ObjectRecoveryMsgbusConsumer.isMsgBusLoaded:
```

Comment:
Can this check be moved inside internalconnect()?

Reply:
Removed the check entirely